velox/functions/prestosql/tests/ProbabilityTest.cpp

```diff
@@ -108,5 +110,52 @@ TEST_F(ProbabilityTest, normalCDF) {
       normal_cdf(0, kNan, 0.1985), "standardDeviation must be > 0");
 }
 
+TEST_F(ProbabilityTest, binomialCDF) {
+  const auto binomialCDF = [&](std::optional<int64_t> numberOfTrials,
+                               std::optional<double> successProbability,
+                               std::optional<int64_t> value) {
+    return evaluateOnce<double>(
+        "binomial_cdf(c0, c1, c2)", numberOfTrials, successProbability, value);
+  };
+
+  EXPECT_EQ(binomialCDF(5, 0.5, 5), 1.0);
+  EXPECT_EQ(binomialCDF(5, 0.5, 0), 0.03125);
+  EXPECT_EQ(binomialCDF(3, 0.5, 1), 0.5);
+  EXPECT_EQ(binomialCDF(20, 1.0, 0), 0.0);
+  EXPECT_EQ(binomialCDF(20, 0.3, 6), 0.60800981220092398);
+  EXPECT_EQ(binomialCDF(200, 0.3, 60), 0.5348091761606989);
+  EXPECT_EQ(binomialCDF(kBigIntMax, 0.5, 2), 0.0);
+  EXPECT_EQ(binomialCDF(kBigIntMax, 0.5, kBigIntMax), 0.0);
+  EXPECT_EQ(binomialCDF(10, 0.5, kBigIntMax), 1.0);
+  EXPECT_EQ(binomialCDF(10, 0.1, kBigIntMin), 0.0);
+  EXPECT_EQ(binomialCDF(10, 0.1, -2), 0.0);
+  EXPECT_EQ(binomialCDF(25, 0.5, -100), 0.0);
+
+  // Invalid inputs
+  VELOX_ASSERT_THROW(
+      binomialCDF(5, -0.5, 3),
+      "successProbability must be in the interval [0, 1]");
+  VELOX_ASSERT_THROW(
+      binomialCDF(5, 2, 3),
+      "successProbability must be in the interval [0, 1]");
+  VELOX_ASSERT_THROW(
+      binomialCDF(5, kBigIntMax, 3),
+      "successProbability must be in the interval [0, 1]");
+  VELOX_ASSERT_THROW(
+      binomialCDF(5, kNan, 3),
+      "successProbability must be in the interval [0, 1]");
+  VELOX_ASSERT_THROW(
+      binomialCDF(-1, 0.5, 2), "numberOfTrials must be greater than 0");
+  VELOX_ASSERT_THROW(
+      binomialCDF(kBigIntMin, 0.5, 1), "numberOfTrials must be greater than 0");
+  VELOX_ASSERT_THROW(
+      binomialCDF(kNan, 0.5, 3), "numberOfTrials must be greater than 0");
```

Comment:
We cannot cast from `kNan` to `long`, ASAN is failing on this line

Reply:
@Yuhta Removed failing test case.